apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/MetadataClientTest.cs

```diff
@@ -120,13 +121,99 @@ public void GetProjectMetadata()
         }
 
         [Fact]
+        public async Task InstanceMetadataChanged()
+        {
+            const string key = "my_instance_key1";
+            const string newValue = "foo";
+
+            var client = MetadataClient.Create();
+            var originalValue = await client.GetCustomInstanceMetadataAsync(key);
+
+            var waitHandle = new ManualResetEventSlim(false);
+            int eventCount = 0;
+            client.InstanceMetadataChanged += (s, e) =>
+            {
+                eventCount++;
+                waitHandle.Set();
+            };
+
+            try
+            {
+                await _fixture.UpdateMetadataAsync($"instance/attributes/{key}", newValue);
+                waitHandle.Wait(TimeSpan.FromSeconds(10));
+                Assert.Equal(1, eventCount);
+                Assert.Equal(newValue, await client.GetCustomInstanceMetadataAsync(key));
+            }
+            finally
```

Comment:
Yes, I think so. I tend to write tests about persistent collections to say "It must contain the one I want" (and maybe "and has grown") rather than having precise expectations, exactly to simplify this sort of thing.